### ISSUE TITLE
Issues with angsd -sites function

### ISSUE THREAD

**Issue by williamamarsh** [ANGSD/angsd#466]
Hi,

I have recently updated my workflow to include conda, and as a result have begun to use angsd installed using conda, in this case f0.935.

However, I no longer get the required output using the command below:

angsd -b [xxxx]  -gl 2 -domajorminor 3 -sites [xxx] -domaf 1 -minmaf 0.05 -doGlf 3 -out test.snp -nthreads 4

which worked perfectly well previously using exactly the same input, with angsd v0.929.

The issue seems to be in the filtering step, as seen below:

for angsd v.0935
```
-> Wed Mar 16 16:27:57 2022
	-> Arguments and parameters for all analysis are located in .arg file
	-> Total number of sites analyzed: 641039227
	-> Number of sites retained after filtering: 0 
	[ALL done] cpu-time used =  581.59 sec
	[ALL done] walltime used =  219.00 sec

```
for angsd v.0929

```
> Mon Oct  4 17:56:14 2021
	-> Arguments and parameters for all analysis are located in .arg file
	-> Total number of sites analyzed: 2710295196
	-> Number of sites retained after filtering: 414575 
	[ALL done] cpu-time used =  3324.33 sec
	[ALL done] walltime used =  3327.00 sec
```
Is this a known issue/how best may I go about remedying it?

Best,

William

**Comment by wenluo711** [comment]
This is so interesting. I'm having another issue and was trying the latest version. Running 2 different versions of angsd using same syntax same input but the latest version gave me 0 sites after filtering while the previous version makes more sense. 
My version upgrade is from `0.935-46-ga3e53f3-dirty` to `0.937` though...

**Comment by zjnolen** [comment]
I can also add another instance to this issue, I've tried using the sites option when creating an SFS with realSFS in both 0.933 and 0.937, and find that if I use it on 0.933, it properly filters to the site list, but on 0.937 it removes all sites.

**Comment by zjnolen** [comment]
After playing a bit with this, it seems to be an issue with the conda install specifically (unless @wenluo711 is seeing it without the conda install). Using 0.937 with htslib 1.15.1 on conda and with an install from source, the `-sites` option works using the source install, but returns 0 sites in the conda install. Conda versions of ANGSD have seemed to have different issues like this when I've tried them in the past (such as #396), but based on #460 there may be an official conda version coming, which potentially could resolve some of these issues. For now, I've switched to using ANGSD in a singularity container rather than conda in my workflows.
